SimCalorimetry/EcalSimProducers/src/EcalDigiProducer.cc

```diff
@@ -310,26 +312,18 @@ void EcalDigiProducer::accumulateCaloHits(HitsHandle const &ebHandle,
 
 void EcalDigiProducer::accumulate(edm::Event const &e, edm::EventSetup const &eventSetup) {
   // Step A: Get Inputs
-  edm::Handle<std::vector<PCaloHit>> ebHandle;
+  const edm::Handle<std::vector<PCaloHit>> &ebHandle = e.getHandle(m_HitsEBToken_);
   if (m_doEB) {
     m_EBShape.setEventSetup(eventSetup);
     m_APDShape.setEventSetup(eventSetup);
-    edm::InputTag ebTag(m_hitsProducerTag, "EcalHitsEB");
-    e.getByLabel(ebTag, ebHandle);
   }
 
-  edm::Handle<std::vector<PCaloHit>> eeHandle;
+  const edm::Handle<std::vector<PCaloHit>> &eeHandle = e.getHandle(m_HitsEEToken_);
   if (m_doEE) {
     m_EEShape.setEventSetup(eventSetup);
-    edm::InputTag eeTag(m_hitsProducerTag, "EcalHitsEE");
-    e.getByLabel(eeTag, eeHandle);
   }
 
-  edm::Handle<std::vector<PCaloHit>> esHandle;
-  if (m_doES) {
-    edm::InputTag esTag(m_hitsProducerTag, "EcalHitsES");
-    e.getByLabel(esTag, esHandle);
-  }
+  const edm::Handle<std::vector<PCaloHit>> &esHandle = e.getHandle(m_HitsESToken_);
 
   accumulateCaloHits(ebHandle, eeHandle, esHandle, 0);
 }
```

Comment:
There are also getByLabel calls left below in https://github.com/cms-sw/cmssw/pull/37773/files#diff-00d232f260111e56efbec17ef741ffab3c9b78dd638786b123215036552b0c50R338 ,
https://github.com/cms-sw/cmssw/pull/37773/files#diff-00d232f260111e56efbec17ef741ffab3c9b78dd638786b123215036552b0c50R344
and https://github.com/cms-sw/cmssw/pull/37773/files#diff-00d232f260111e56efbec17ef741ffab3c9b78dd638786b123215036552b0c50R350

Reply:
This is left because PileUpEvent has no method to access collection using token. One has to work at the level of SimGeneral/MixingModule/interface/PileUpEventPrincipal.h